integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java

```diff
@@ -25,26 +25,75 @@
 class ValidatorTest {
 
     @Test
-    public void validXML() {
+    public void validXMLFromClassPath() {
 
         RestAssured.given()
                 .contentType(ContentType.XML)
                 .body("<message><firstName>MyFirstname</firstName><lastName>MyLastname</lastName></message>")
-                .post("/validator/xml")
+                .post("/validator/classpath")
                 .then()
                 .statusCode(200);
 
     }
```

Comment:
I'd vote for having one parameterized test for the valid case and one for invalid case. The params being `"classpath", "filesystem", "http"`. Just grep for `@ParameterizedTest` in the source tree to see some examples.
Less code is perhaps easier to maintain.